mail_activity_plan/models/mail_activity_mixin.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from lxml import etree
+
+from odoo import _, api, fields, models
+
+from odoo.addons.base.models import ir_ui_view
+
+
+class MailActivityMixin(models.AbstractModel):
+    _inherit = "mail.activity.mixin"
+
+    has_activity_plans = fields.Boolean(
+        compute="_compute_has_activity_plans",
+    )
+
+    def _compute_has_activity_plans(self):
+        plan_model = self.env["mail.activity.plan"]
+        for item in self:
+            plans = plan_model._get_plans_from_model(item._name)
+            item.has_activity_plans = True if len(plans) else False
+
+    def _get_mail_activity_plan_button_node(self):
+        attrs = {"invisible": [("has_activity_plans", "=", False)]}
+        return etree.Element(
+            "button",
+            attrib={
+                "name": "action_wizard_mail_activity_plan",
+                "type": "object",
+                "class": "btn-secondary",
+                "attrs": str(attrs),
+                "string": _("Launch Activity Plan"),
+            },
+        )
+
+    def _get_mail_activity_plan_field_node(self):
+        return etree.Element(
+            "field", attrib={"name": "has_activity_plans", "invisible": "1"}
+        )
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):
+        """Add (inside <header>) the has_activity_plans field and button to launch
+        activity plans."""
```

Comment:
And remember that a multi-line docstring put the ending 3 quotes in a new line.

```suggestion
        activity plans, creating the element if not exists.
        """
```